packages/console/src/VncConsole/VncConsole.js

```diff
@@ -1,11 +1,180 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-const propTypes = {};
-const defaultProps = {};
+import * as NovncLog from '@novnc/novnc/core/util/logging';
+import RFB from '@novnc/novnc/core/rfb';
 
-const VncConsole = () => <div>VncConsole</div>;
+import { noop } from 'patternfly-react';
 
-VncConsole.propTypes = propTypes;
-VncConsole.defaultProps = defaultProps;
+import VncActions from './VncActions';
+
+const CONNECTING = 'connecting';
+const CONNECTED = 'connected';
+const DISCONNECTED = 'disconnected';
```

Comment:
afair there are more states? e.g. https://github.com/theforeman/foreman/blob/develop/app/assets/javascripts/noVNC.js#L14

Reply:
These states reflect events received from the RFB (https://github.com/novnc/noVNC/blob/master/docs/API.md#events).

This seems to be changed as well.
For example, in https://github.com/novnc/noVNC/blob/0342e4f4891a0c29f10b25a53258bf40890bc9fd/core/rfb.js#L675 , the failure is wrapped behind the `disconnected` state.